### ISSUE TITLE
Milliseconds shown twice when using custom full date format

### ISSUE THREAD

**Issue by klehelley** [grafana/grafana#52405]
**What happened**:

When viewing Elasticsearch data with the help of a Table panel, the date-time value is shown with the sub-seconds appearing twice:

![2022-07-18 15 28 11 grafana admin preprod dpsc 14b2217e8168](https://user-images.githubusercontent.com/11931306/179524946-bb99646d-a42f-4540-8fb4-ce9ff64e133d.png)

When inspecting the table and disabling "Formatting data", the date-time is shown in the expected format (albeit in UTC, which we use when storing the documents):

![2022-07-18 15 29 00 grafana admin preprod dpsc 84bc435463c0](https://user-images.githubusercontent.com/11931306/179525231-b255af59-13fd-416c-82e2-0c56c0c30785.png)

The date-time format we want to use was configured globally with the help of the environment variable `GF_DATE_FORMATS_FULL_DATE` (set to `YYYY-MM-DDTHH:mm:ss.SSSZ`).

**What you expected to happen**:

The date-time should be shown in the expected format, that is with the sub-seconds appearing only once.

**How to reproduce it (as minimally and precisely as possible)**:

Run Grafana with the following environment variable set:

```sh
GF_DATE_FORMATS_FULL_DATE=YYYY-MM-DDTHH:mm:ss.SSSZ
```

Connect to an Elasticsearch instance containing documents with a date field containing values such as `2022-07-18T13:55:18.383Z`.

View the documents via Grafana Explore or using a Table panel in a dashboard.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.4
- Data source type & version: Elasticsearch 7.10.2 (Open Distro 1.13.3)
- OS Grafana is installed on: Docker image derived from grafana/grafana:8.4.4 (embeds some custom configuration and dashboards, but nothing that is expected to have an impact on the behavior described above from what we understand)
- User OS & Browser: Ubuntu 22.04 LTS & Vivaldi 5.2.2623.46, Ubuntu 20.04 LTS & Firefox 102.0
- Grafana plugins: 42 Grafana Labs plugins including "Table" and "Table (old)" (only "Table" appears in the panel dropdown menu) + ClickHouse
- Others:

**Comment by ashharrison90** [comment]
hi @klehelley 👋 

it's possible this has been fixed already as i can't reproduce this locally on the latest version of grafana.
in order to double check, are you able to provide the results from the query inspector? 

you can get this from `query inspector` -> `query tab` -> hit `refresh` -> `copy to clipboard`

feel free to redact any sensitive information - the important thing is the frame data in the response.

thanks! 🙌

**Comment by klehelley** [comment]
Sorry for the delay, I have finally had time today to perform more tests following your response. We have been able to reproduce the issue with the following Grafana versions: 8.5.0, 8.5.10 and 9.0.2. Below is an extract of the data that is being retrieved by Grafana, as shown in the query inspector:

```json
{
  "request": {
    "url": "api/datasources/proxy/1/_msearch",
    "method": "POST",
    "data": "{\"search_type\":\"query_then_fetch\",\"ignore_unavailable\":true,\"index\":\"kast-*\"}\n{\"size\":500,\"query\":{\"bool\":{\"filter\":[{\"range\":{\"@timestamp\":{\"gte\":1660037190211,\"lte\":1660058790211,\"format\":\"epoch_millis\"}}},{\"query_string\":{\"analyze_wildcard\":true,\"query\":\"kubernetes.namespace_name.keyword:* AND kubernetes.pod_name.keyword:*\"}}]}},\"sort\":[{\"@timestamp\":{\"order\":\"desc\",\"unmapped_type\":\"boolean\"}},{\"_doc\":{\"order\":\"desc\"}}],\"script_fields\":{}}\n",
    "hideFromInspector": false
  },
  "response": {
    "took": 10,
    "responses": [
      {
        "took": 10,
        "timed_out": false,
        "_shards": {
          "total": 1,
          "successful": 1,
          "skipped": 0,
          "failed": 0
        },
        "hits": {
          "total": {
            "value": 147,
            "relation": "eq"
          },
          "max_score": null,
          "hits": [
            {
              "_index": "idx-2022.08.09",
              "_type": "_doc",
              "_id": "FeI2g4IBZ6kPH2KbrTMt",
              "_score": null,
              "_source": {
                "@timestamp": "2022-08-09T15:26:13.830Z",
                "kubernetes": {
                  "host": "worker",
                  "container_name": "elasticsearch",
                  "docker_id": "043c9e716f23b75fa3216c74cabd5b5e92867616b1b9ac508d7e982718613e99",
                  "pod_name": "elasticsearch-0",
                  "namespace_name": "doc-store",
                  "pod_id": "1b9a812c-d966-471a-baa4-dfc05dd9afc3"
                },
                "log_processed": {
                  "node_name": "elasticsearch-0",
                  "type": "server",
                  "cluster_uuid": "7oYkh4r5QXiKpOV7ZNDGOA",
                  "node_id": "QmMaHx2cTKu0s42P3qT3Zw",
                  "message": "[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.",
                  "component": "o.e.d.s.a.b.h.DateHistogramAggregationBuilder",
                  "level": "DEPRECATION",
                  "cluster_name": "elasticsearch-doc-store"
                },
                "stream": "stdout",
                "logtag": "F"
              },
              "sort": [
                1660058773830,
                25491
              ]
            },
            {
              "_index": "idx-2022.08.09",
              "_type": "_doc",
              "_id": "5uI2g4IBZ6kPH2KbmTKl",
              "_score": null,
              "_source": {
                "@timestamp": "2022-08-09T15:26:09.672Z",
                "kubernetes": {
                  "host": "worker",
                  "container_name": "elasticsearch",
                  "docker_id": "043c9e716f23b75fa3216c74cabd5b5e92867616b1b9ac508d7e982718613e99",
                  "pod_name": "elasticsearch-0",
                  "namespace_name": "doc-store",
                  "pod_id": "1b9a812c-d966-471a-baa4-dfc05dd9afc3"
                },
                "log_processed": {
                  "node_name": "elasticsearch-0",
                  "type": "server",
                  "cluster_uuid": "7oYkh4r5QXiKpOV7ZNDGOA",
                  "node_id": "QmMaHx2cTKu0s42P3qT3Zw",
                  "message": "[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.",
                  "component": "o.e.d.s.a.b.h.DateHistogramAggregationBuilder",
                  "level": "DEPRECATION",
                  "cluster_name": "elasticsearch-doc-store"
                },
                "stream": "stdout",
                "logtag": "F"
              },
              "sort": [
                1660058769672,
                25493
              ]
            }
          ]
        },
        "status": 200
      }
    ],
    "$$config": {
      "url": "api/datasources/proxy/1/_msearch",
      "method": "POST",
      "data": "{\"search_type\":\"query_then_fetch\",\"ignore_unavailable\":true,\"index\":\"kast-*\"}\n{\"size\":500,\"query\":{\"bool\":{\"filter\":[{\"range\":{\"@timestamp\":{\"gte\":1660037190211,\"lte\":1660058790211,\"format\":\"epoch_millis\"}}},{\"query_string\":{\"analyze_wildcard\":true,\"query\":\"kubernetes.namespace_name.keyword:* AND kubernetes.pod_name.keyword:*\"}}]}},\"sort\":[{\"@timestamp\":{\"order\":\"desc\",\"unmapped_type\":\"boolean\"}},{\"_doc\":{\"order\":\"desc\"}}],\"script_fields\":{}}\n",
      "hideFromInspector": false
    }
  }
}
```

I have tried to study the front-end code via the browser tools, since it seems to be the culprit, unfortunately I haven't been able to make an in-depth study.

I just noticed something in the JSON data I have pasted. `"unmapped_type":"boolean"` for `@timestamp`? Looks fishy to me. Could it be the cause of the issue?

**Comment by klehelley** [comment]
Additionally, tests were also performed with fields containing millisecond timestamp values. The field `"timestampMs": 1659460121976` returned by Elasticsearch is shown as `2022-08-02T19:08:41.976.976+02:00` in the table. The Elasticsearch response also shows `"unmapped_type":"boolean"` for this field in the `request` and `$$config` fields.

**Comment by klehelley** [comment]
Hello, has anyone had time to look into our issue? Is the additional information provided in my last message useful (I am in no way an Elasticsearch nor Grafana expert, so only making a supposition here)? Is there additional information we can provide, or tests we can perform to help understand the issue and identify a fix (be it in our configuration or in Grafana)?

**Comment by zuchka** [comment]
@klehelley 

I can confirm that this bug appears on 8.4.4 but it appears fixed on `Grafana v9.2.5 (042e4d216b)` and `v9.3.0-88845pre (d33939da55)`.

Closing as fixed. @klehelley I hope upgrading will get you unblocked on this.
